notifications/manager.go

```diff
@@ -0,0 +1,214 @@
+package notifications
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"time"
+
+	"github.com/lightninglabs/loop/swapserverrpc"
+)
+
+type NotificationType int
+
+const (
+	NotificationTypeUnknown NotificationType = iota
+	NotificationTypeReservation
+)
+
+// Config contains all the services that the notification manager needs to
+// operate.
+type Config struct {
+	// Client is the client used to communicate with the swap server.
+	Client swapserverrpc.SwapServerClient
+
+	// FetchL402 is the function used to fetch the l402 token.
+	FetchL402 func(context.Context) error
+}
+
+// Manager is a manager for notifications that the swap server sends to the
+// client.
+type Manager struct {
+	cfg *Config
+
+	hasL402 bool
+
+	runCtx context.Context
+
+	subscribers map[NotificationType]map[string]func(*swapserverrpc.SubscribeNotificationsResponse) //nolint:lll
+}
+
+// NewManager creates a new notification manager.
+func NewManager(cfg *Config) *Manager {
+	return &Manager{
+		cfg:         cfg,
+		subscribers: make(map[NotificationType]map[string]func(*swapserverrpc.SubscribeNotificationsResponse)), //nolint:lll
+	}
+}
+
+// SubscribeReservations subscribes to the reservation notifications.
+func (n *Manager) SubscribeReservations(
```

Comment:
I suggest to change the interface to accept context and reservationResChan.

Then we don't need a callback, ID, and Unsubscribe method.